security_monkey/watchers/direct_connect/virtual_gateway.py

```diff
@@ -0,0 +1,98 @@
+#     Copyright 2016 Bridgewater Associates
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+"""
+.. module: security_monkey.watchers.direct_connect.virtual_gateway
+    :platform: Unix
+
+.. version:: $$VERSION$$
+.. moduleauthor:: Bridgewater OSS <opensource@bwater.com>
+
+
+"""
+from security_monkey.watcher import Watcher
+from security_monkey.watcher import ChangeItem
+from security_monkey.constants import TROUBLE_REGIONS
+from security_monkey.exceptions import BotoConnectionIssue
+from security_monkey import app
+from boto.directconnect import regions
+
+
+class VirtualGateway(Watcher):
+    index = 'virtual_gateway'
+    i_am_singular = 'Virtual Gateway'
+    i_am_plural = 'Virtual Gateways'
+
+    def __init__(self, accounts=None, debug=False):
+        super(VirtualGateway, self).__init__(accounts=accounts, debug=debug)
+
+    def slurp(self):
+        """
+        :returns: item_list - list of virtual gateways
+        :returns: exception_map - A dict where the keys are a tuple containing the
+            location of the exception and the value is the actual exception
+
+        """
+        self.prep_for_slurp()
+        from security_monkey.common.sts_connect import connect
+        item_list = []
+        exception_map = {}
+        for account in self.accounts:
+            for region in regions():
+                app.logger.debug(
+                    "Checking {}/{}/{}".format(self.index, account, region.name))
+                try:
+                    dc = connect(account, 'boto3.ec2.client', region=region)
+                    response = self.wrap_aws_rate_limited_call(
+                        dc.describe_vpn_gateways
```

Comment:
The file direct_connect/virtual_gateway.py:
- does not call `describe_virtual_gateways()` as you might expect with a 'directconnect' boto3 client
- instead calls `describe_vpn_gateways()` with an `ec2` boto3 client

Should probably be renamed to vpn_gateways.py and maybe placed in the ec2 or vpc folder.